File: vaccin-vaccine/src/main/java/com/unufolio/vaccin/vaccine/dataobject/BaseEntity.java
package com.unufolio.vaccin.vaccine.dataobject;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;

import java.time.LocalDateTime;

/**
 * @author Unufolio unufolio@gmail.com
 * @since 2022/03/29
 */
public class BaseEntity {
    /**
     * 主键
     */
    @TableId(type = IdType.AUTO)
    private Long id;
    /**
     * 创建时间
     */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
    /**
     * 创建时间
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime modifyTime;
    /**
     * 是否删除
     */
    private Boolean isDeleted;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public LocalDateTime getCreateTime() {
        return createTime;
    }

    public void setCreateTime(LocalDateTime createTime) {
        this.createTime = createTime;
    }

    public LocalDateTime getModifyTime() {
        return modifyTime;
    }

    public void setModifyTime(LocalDateTime modifyTime) {
        this.modifyTime = modifyTime;
    }

    public Boolean getDeleted() {
        return isDeleted;
    }

    public void setDeleted(Boolean deleted) {
        isDeleted = deleted;
    }

    @Override
    public String toString() {
        return "BaseDataObject{" +
                "id=" + id +
                ", createTime=" + createTime +
                ", modifyTime=" + modifyTime +
                ", isDeleted=" + isDeleted +
                '}';
    }
}

File: vaccin-vaccine/src/main/kotlin/com/unufolio/vaccin/vaccine/dataobject/BaseEntity.kt
package com.unufolio.vaccin.vaccine.dataobject

import com.baomidou.mybatisplus.annotation.FieldFill
import com.baomidou.mybatisplus.annotation.IdType
import com.baomidou.mybatisplus.annotation.TableField
import com.baomidou.mybatisplus.annotation.TableId
import java.time.Instant

/**
 * @author Unufolio unufolio@gmail.com
 * @since 2022/03/29
 */
abstract class BaseEntity(
    /**
     * 主键
     */
    @TableId(type = IdType.AUTO)
    var id: Long? = null,
    /**
     * 创建时间
     */
    @TableField(fill = FieldFill.INSERT)
    var createTime: Instant? = null,
    /**
     * 创建时间
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    var modifyTime: Instant? = null,
    /**
     * 是否删除
     */
    var isDeleted: Boolean? = null
)

/**
 * @author Unufolio unufolio@gmail.com
 * @since 2022/03/29
 */
abstract class TenantBaseEntity(
    var tenantId: Long? = null,
    /**
     * super
     */
    id: Long?,
    createTime: Instant?,
    modifyTime: Instant?,
    isDeleted: Boolean?
) : BaseEntity(id, createTime, modifyTime, isDeleted) {

    override fun toString(): String {
        return "TenantBaseEntity(tenantId=$tenantId) ${super.toString()}"
    }
}